### ISSUE TITLE
[YSQL] large number of error due to revoke command

### ISSUE THREAD

**Issue by myang2021** [yugabyte/yugabyte-db#12528]
Jira Link: [[DB-398]](https://yugabyte.atlassian.net/browse/DB-398)
### Description

From slack:
We have a customer PoC running and they are having a large number of errors:
`ERROR: Query error: The catalog snapshot used for this transaction has been invalidated.`

After turning on ddl auditing, it was found that there was a revoke command executed after a new
database was created.

```
2022-05-10 06:53:25.750 UTC [15530] LOG:  statement: REVOKE ALL ON schema public FROM public ;

```

Currently, REVOKE is considered as a breaking change, as shown below:
```
        // T_Grant...
        case T_GrantStmt:
        {
            /* Grant (add permission) is not a breaking change, but revoke is. */
            GrantStmt *stmt = castNode(GrantStmt, parsetree);
            *is_breaking_catalog_change = !stmt->is_grant;
            break;
        }
```
REVOKE or GRANT are both represented as a T_GrantStmt. If stmt->is_grant is false, it means REVOKE.
As we can see from above, REVOKE is a breaking change which means the `last_breaking_version` column
in `pg_yb_catalog_version` table will be updated to a newer version. All running statements that
were prepared using the previous catalog version will be smaller than the new version and therefore
will be detected and then cause the statement’s enclosing transaction to abort.

More on the customer's use case:

We are building a multi-tenant platform. When a customer creates a so called project we create a database for that project. When they deploy a service we create a schema and role (user) within that database. They can deploy many services in the same projects, i.e. we create new schemas in the database.

The db role (user) should only have access to its own schema.

The sequence of the statements is something like
CREATE DATABASE db1;
\c db1
REVOKE ALL ON schema public FROM public;
REVOKE connect ON DATABASE db1 FROM public;

There are 2 REVOKE statements. The second REVOKE involves an update of pg_database table which is a
shared table. Therefore the per-database catalog version will not help to solve the catalog `breaking
change` problem because in the general case if a shared catalog table is updated, then any PG cache
that stored the shared catalog table is now stale and needs to be refreshed. We also needs to
pessimistically assume any ongoing transaction may involve this shared catalog table and therefore
should be aborted for correct semantics.

[DB-398]: https://yugabyte.atlassian.net/browse/DB-398?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by myang2021** [comment]
A work around is to add a session variable such as `yb_enable_nonbreaking_ddl_mode` that advanced user can use to simply make the next ddl statement non-breaking. In this case if the user is fully aware that the revoke is only related to a newly created database for their multi-tenancy support, they can do
```
set yb_enable_nonbreaking_ddl_mode = true;
REVOKE ALL ON schema public FROM public;  
REVOKE connect ON DATABASE db1 FROM public;
set yb_enable_nonbreaking_ddl_mode = false;
```

**Comment by myang2021** [comment]
fixed by 47fcdf834b0c12644c8c5e405363a96f1d47a519
